storage/remote/client.go

```diff
@@ -144,8 +143,19 @@ type ReadClient interface {
 	Read(ctx context.Context, query *prompb.Query, sortSeries bool) (storage.SeriesSet, error)
 }
 
+func registerRemoteReadMetrics(registry prometheus.Registerer) {
+	once.Do(func() {
```

Comment:
As commented elsewhere, we cannot use this `Once.Do` pattern without defeating the purpose of removing the global registry in the first place.